x-pack/platform/packages/shared/onechat/onechat-common/chat/events.ts

```diff
@@ -126,8 +126,8 @@ export const isMessageChunkEvent = (
 export interface MessageCompleteEventData {
   /** ID of the message */
   message_id: string;
-  /** full text content of the message */
-  message_content: string;
+  /** full text content of the message, or structured data object when structuredOutput is true */
+  message_content: string | object;
```

Comment:
Given those events are used up to the UI, it's going to force us adapt things here. 

Do you think it's better than adding an optional `structured_output` to `MessageCompleteEventData` as we did for the assistant response? Feels like the new optional field would require less changes in other parts of the code, wdyt?

(out of scope, but tbh the "concept" of "message events" itself is kinda outdated, so maybe I should just rewrite that part, idk)